torch/linalg/__init__.py

```diff
@@ -1609,6 +1609,85 @@
     tensor(2.4576e-07)
 """)
 
+svd_rank_restricted = _add_docstr(_linalg.linalg_svd_rank_restricted, r"""
+linalg.svd_rank_restricted(A, *, atol=None, rtol=None, out=None) -> (Tensor, Tensor, Tensor, Tensor)
+
+Computes a singular value decomposition (SVD) and the numerial rank of a matrix.
+The SVD decomposition will be of the numerical rank of :attr:`A` with the non-rank
+contributing rows/columns/entries set to zero.
+
+The matrix rank is computed as the number of singular values
+that are greater than :math:`\max(\text{atol}, \sigma_1 * \text{rtol})` threshold,
+where :math:`\sigma_1` is the largest singular value (or eigenvalue).
+
+Supports input of float, double, cfloat and cdouble dtypes.
+Also supports batches of matrices, and if :attr:`A` is a batch of matrices then
+the output has the same batch dimensions.
+
+If :attr:`rtol` is not specified and :attr:`A` is a matrix of dimensions `(m, n)`,
+the relative tolerance is set to be :math:`\text{rtol} = \max(m, n) \varepsilon`
+and :math:`\varepsilon` is the epsilon value for the dtype of :attr:`A` (see :class:`.finfo`).
+If :attr:`rtol` is not specified and :attr:`atol` is specified to be larger than zero then
+:attr:`rtol` is set to zero.
+
+If :attr:`atol` or :attr:`rtol` is a :class:`torch.Tensor`, its shape must be broadcastable to that
+of the singular values of :attr:`A` as returned by :func:`torch.linalg.svd`.
+
+.. note::
+    It is advised to use this function if gradients for singular :attr:`A` are needed.
+    The gradient computation only operates over the first singular subspaces of :attr:`A` of size :attr:`rank`,
+    which increases performance and also eliminates the issue of numerically unstable gradients with repeated zero singular values
+    that the backward for :func:`torch.linalg.svd` has.
```

Comment:
What about simply saying:
> This function is differentiable when the matrix :attr:`A` is singular (for example, when `A` is low-rank) or close to singular, while :func:`~linalg.svd` is not.
It is also expected to be more performant than :func:`~linalg.svd` for low-rank inputs.

This way we don't claim performance benefits in the large, but just in the low-rank case.